### ISSUE TITLE
problem running train.py

### ISSUE THREAD

**Issue by aspenlin** [ildoonet/tf-pose-estimation#486]
I'm running train.py with the following arguments:

--model=cmu, --batchsize=16, --gpus=1, --max-epoch=2, --input-width=432, --input-height=368

but it is running super slow, from my estimation, this will take up to at least 10 hours although it's only 2 epochs.
I check the GPU usage with 'nvidia-smi', and it seems the usage is low, below is the result:

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 418.40.04    Driver Version: 418.40.04    CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla K80           On   | 00000000:00:1E.0 Off |                    0 |
| N/A   64C    P0    66W / 149W |     67MiB / 11441MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0     18783      C   python3                                       56MiB |
+-----------------------------------------------------------------------------+

one other problem is as you can see the CUDA version shown above is 10.1, but nvcc -V gives:
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2018 NVIDIA Corporation
Built on Wed_Apr_11_23:16:29_CDT_2018
Cuda compilation tools, release 9.2, V9.2.88

which means the Cuda version is 9.2??

I also have the below message shown repeated for more than 10 times when running train.py:

2019-07-17 18:45:05.022192: W tensorflow/core/common_runtime/colocation_graph.cc:1016] Failed to place the graph without changing the devices of some resources. Some of the operations (that had to be colocated with resource generating operations) are not supported on the resources' devices. Current candidate devices are [
  /job:localhost/replica:0/task:0/device:CPU:0].
See below for details of this colocation group:
Colocation Debug Info:
Colocation group had the following types and supported devices: 
Root Member(assigned_device_name_index_=-1 requested_device_name_='/device:GPU:0' assigned_device_name_='' resource_device_name_='/device:GPU:0' supported_device_types_=[CPU] possible_devices_=[]
Assign: CPU 
ApplyAdam: CPU 
RandomUniform: CPU XLA_CPU XLA_GPU 
Const: CPU XLA_CPU XLA_GPU 
Mul: CPU XLA_CPU XLA_GPU 
Sub: CPU XLA_CPU XLA_GPU 
Add: CPU XLA_CPU XLA_GPU 
Identity: CPU XLA_CPU XLA_GPU 
VariableV2: CPU 

Colocation members, user-requested devices, and framework assigned devices, if any:
  conv1_1/biases/Initializer/random_uniform/shape (Const) 
  conv1_1/biases/Initializer/random_uniform/min (Const) 
  conv1_1/biases/Initializer/random_uniform/max (Const) 
  conv1_1/biases/Initializer/random_uniform/RandomUniform (RandomUniform) 
  conv1_1/biases/Initializer/random_uniform/sub (Sub) 
  conv1_1/biases/Initializer/random_uniform/mul (Mul) 
  conv1_1/biases/Initializer/random_uniform (Add) 
  conv1_1/biases (VariableV2) /device:GPU:0
  conv1_1/biases/Assign (Assign) /device:GPU:0
  conv1_1/biases/read (Identity) /device:GPU:0
  conv1_1/biases/Adam/Initializer/zeros (Const) /device:GPU:0
  conv1_1/biases/Adam (VariableV2) /device:GPU:0
  conv1_1/biases/Adam/Assign (Assign) /device:GPU:0
  conv1_1/biases/Adam/read (Identity) /device:GPU:0
  conv1_1/biases/Adam_1/Initializer/zeros (Const) /device:GPU:0
  conv1_1/biases/Adam_1 (VariableV2) /device:GPU:0
  conv1_1/biases/Adam_1/Assign (Assign) /device:GPU:0
  conv1_1/biases/Adam_1/read (Identity) /device:GPU:0
  Adam/update_conv1_1/biases/ApplyAdam (ApplyAdam) /device:GPU:0
  save/Assign_423 (Assign) /device:GPU:0
  save/Assign_424 (Assign) /device:GPU:0
  save/Assign_425 (Assign) /device:GPU:0

I'm new to machine learning. Can anyone explain why I'm having the above results? Is 10 hour normal for 2 epoch in a machine with 60GB memory and one GPU? Thanks a lot for any help.

**Comment by aspenlin** [comment]
Sorry I was wrong about the time estimation. I spent half a day and it's only 0.06 epoch done. So it will take me 8 days to complete just 2 epochs. Although I only have 1 GPU and 60GB memory, there must be something wrong here, am I right? 
Can someone give me some idea about how to debug? I believe either my GPU nor memory is fully utilized.
Thanks for any help.

**Comment by aspenlin** [comment]
<img width="723" alt="Screen Shot 2019-07-18 at 10 30 11" src="https://user-images.githubusercontent.com/22419710/61465993-2335a080-a947-11e9-8c86-72b7762421ee.png">

I realized that the GPU-util is 0%?? Anybody has any idea why this is happening? My training is ongoing and it has some output although it is really slow... It has taken me more than 15 hours to achieve 0.06 epoch as attached. My batchsize is 16 and learning rate 0.001, with cmu model

![Screen Shot 2019-07-18 at 10 32 38](https://user-images.githubusercontent.com/22419710/61466217-845d7400-a947-11e9-8516-d71046990f80.png)

**Comment by aspenlin** [comment]
My 4 CPU are 100% utilized. Is this CPU bottleneck causing the GPU utilization to be 0??

**Comment by aspenlin** [comment]
Problem solved. The training is much faster now after I installed necessary cuda libraries and cudnn. It turns out GPU was not really used before because tensorflow cannot find some libraries, although it can still train.

**Comment by xiaopingxiao** [comment]
> Problem solved. The training is much faster now after I installed necessary cuda libraries and cudnn. It turns out GPU was not really used before because tensorflow cannot find some libraries, although it can still train.
hello，i want to recover the effects of the paper, but I'm running the train.py Error .
i have only one gpu,but gpus default is 4,i dont understand that have other problem.
I'm using coco2014 data set, the same as others. Can you help me?

**Comment by deploredeveloper** [comment]
What libraries would be required? I am having the same error running Tensorflow default benchmark

**Comment by tbbjymm21** [comment]
> What libraries would be required? I am having the same error running Tensorflow default benchmark

You should install cuda libraries and cudnn(Any version will do). The most important thing is to install tensorflow-gpu!!!
